oauth.rst

```diff
@@ -1,131 +1,129 @@
+- ``authorization_code``: this is the code that was created when the user approved to grant you access. It expires after a few minutes, so you are expected to use it right away to get an Access Token.
 
-Basic Client Integration Flow
-=============================
-Clients request an ``authorization_code`` grant type by redirecting the user to the Dock Authorization Page. The user verifies the information the Client app is requesting (scope) and also see the app's name, logo and description. After approval, the user is redirected to the ``return_uri`` so the Client gets an ``authorization code`` which it can then exchange for an ``access token`` as explained before. It is mandatory that this last call takes place in the Client's backend to avoid MITM attacks. Dock will only return an ``access token`` to the right ``authorization code`` coming from the right Client, using the right client credentials.
-``dock-auth`` accepts Bearer Tokens in the ``Authorization`` header. To make sense of this it can be useful to observe how a sample step-by-step flow would look like for a client getting an Access Token to act on behalf of one of its users, and then using it to get user data:
 
-1) Authorization grant query
-----------------------------
-The client needs to start an authorization grant query by redirecting the user to:
-``GET https://app.dock.io/oauth/authorize`` with the following url params:
 
-- ``client_id``: id of the Client or third party app doing the request
-- ``response_type=code``: 'code' is always expected. It means that the Client expects an ``authorization_code`` back as response
-- ``redirect_uri``: URL where the Client wants the user to be redirected by the Authorization Server if the authorization grant query is successful
-- ``scope``: scope name(s). In short it defines which of the user's resources the Client will be getting access to. Our api expects comma-separated values if more than one.
-- ``state``: (optional) useful for the client to store encoded info. Will be returned back to the Client unmodified.
+Step 3 - Using an Authorization Code to get an Access Token
+-----------------------------------------------------------
 
-2) Authorization page
----------------------
-The user is presented with a page where he needs to:
+You now have an ``authorization_code`` that you can use on a call to
+``GET https://app.dock.io/api/v1/oauth/access-token`` to get the actual ``access token``. This is the Token that is to be stored for this User in your application. Remember: this call needs to be made from the BE so the traffic is not visible in the User's browser. The following params are expected in the URL for this call:
 
-i) Authenticate (if he hasn't already done so)
-ii) Check the Application data (client name, description, logo, etc)
-iii) Approve the requested access to the given scope
+- ``grant_type``: ``authorization_code`` is expected by default.
```

Comment:
I would use "authorization_code" with quotes instead of ``authorization_code`` since that's the notation we use for parameters. This is a constant that should contain the word "authorization_code"